src/vm/fieldmarshaler.cpp

```diff
@@ -3364,14 +3364,22 @@ VOID FieldMarshaler_FixedStringAnsi::UpdateNativeImpl(OBJECTREF* pCLRValue, LPVO
             nc = m_numchar - 1;
```

Comment:
I think this check is fundamentally incorrect - it did not consider multi-byte ANSI char at all and I believe we should simply let InternalWideToAnsi do the right thing.